cupsfilters/pdftopdf/pdftopdf_processor.cc

```diff
@@ -30,6 +30,17 @@ bool ProcessingParameters::withPage(int outno) const // {{{
   return pageRange.contains(outno);
 }
 // }}}
+bool ProcessingParameters::havePage(int pageno) const
+{
+  if (pageno%2 == 0) { // 1-based
+    if (!evenPages) {
+      return false;
+    }
+  } else if (!oddPages) {
+    return false;
+  }
+  return origPageRange.contains(pageno);
+}
 
```

Comment:
Note that page-set=even and page-set=odd applies to the output pages as the original page-ranges attribute. The page-set attribute is for manual duplex (print odd pages, turn over the stack, put back into input tray, and print even pages). orig-page-ranges should simply select the pages from the input stream and the resulting list of pages passed on for all the other functionality of pdftopdf() to be applied.
If you have a 20-page input document and apply "orig-page-ranges=1-5,7-8,13-15 number-up=2 page-set=odd page-ranges=1-4" you get 2 sheets, the first contains the original pages 1 and 2, the second contains the original pages 5 and 7. Note that page-set and page-ranges are applied simultaneously. The page must be selected by both page-set and page-ranges to get finally selected. In our example we select 10 pages from the original 20 pages, then we put two on each sheet, ending up with 5 sheets, containing 1+2, 3+4, 5+7, 8+13, 14+15, after that we want to have the sheets 1, 2, 3,, and 4 but only the odd ones, so it remains sheet 1 and sheet 3, containing 1+2 and 5+7.

Reply:
Thanks for bringing this to my attention.I made the necessary adjustments, and the option now functions as you indicated.